File: Arrays & HashMap/Valid Anagram/Valid Anagram.kt
class Solution {
    fun isAnagram(s: String, t: String): Boolean {
        if (s.length != t.length) return false

        val map = mutableMapOf<Char, Int>()

        s.forEach { c -> 
            map[c] = map.getOrDefault(c, 0) + 1
        } 

        t.forEach { c -> 
            map[c] = map.getOrDefault(c, 0) - 1
        }

        return map.values.all {
            it == 0
        }        
    }
}

//Time Complexity: O(n)
//Space Complexity: O(n)

File: Arrays & HashMap/Valid Anagram/Valid Anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        if len(s) != len(t):
            return False
        
        map = defaultdict(int)

        for c in s:
            map[c] += 1
        
        for c in t:
            map[c] -= 1
        
        return all(value == 0 for value in map.values())

'''
Time Complexity: O(n)
Space Complexity: O(n)
'''
